cmd/istioctl/istioctl_test.go

```diff
@@ -15,159 +15,359 @@
 package main
 
 import (
+	"errors"
+	"fmt"
 	"testing"
 
-	"istio.io/manager/test/mock"
+	"strings"
+
+	"istio.io/manager/apiserver"
+	"istio.io/manager/model"
 )
 
-func rootSetup(t *testing.T) {
-	config = mock.MakeRegistry()
+type StubClient struct {
```

Comment:
I like the idea in theory but it gives us less control at the unit test level and introduces a bit of test scope leakage. If we update the config registry interface it then may also break these tests.